lean4-infoview/src/infoview/tooltips.tsx

```diff
@@ -70,15 +64,31 @@ export function Tooltip(props_: TooltipProps) {
     return ReactDOM.createPortal(floating, document.body)
 }
 
-export const WithToggleableTooltip = forwardAndUseStateRef<
-    HTMLSpanElement,
-    React.HTMLProps<HTMLSpanElement> & {
-        isTooltipShown: boolean
-        hideTooltip: () => void
-        tooltipChildren: React.ReactNode
+export interface ToggleableTooltip {
+    tooltip: JSX.Element
+    tooltipDisplayed: boolean
+    setTooltipDisplayed: (tooltipDisplayed: boolean) => void
+    onClick: () => void
+    onClickOutside: () => void
+}
+
+/**
+ * Provides handlers to show a tooltip when a state variable is changed. Hides the tooltip when clicking on it our outside of it.
+ */
+export function useToggleableTooltip(
+    ref: React.RefObject<HTMLSpanElement>,
+    tooltipChildren: React.ReactNode,
+): ToggleableTooltip {
+    const [anchor, setAnchor] = React.useState<HTMLSpanElement | null>(null)
+    const [tooltipDisplayed, setTooltipDisplayed_] = React.useState<boolean>(false)
+    const setTooltipDisplayed = (tooltipDisplayed: boolean) => {
+        setTooltipDisplayed_(tooltipDisplayed)
+        if (tooltipDisplayed) {
+            // Setting the tooltip anchor lazily only when the tooltip is displayed avoids accidental
+            // re-renders induced by setting this state variable during the initial render of a component.
+            setAnchor(ref.current)
```

Comment:
I think I tried this at some point and it stopped working correctly. The floating-ui docs are pretty explicit that this needs to be state so that it is re-rendered correctly.